### ISSUE TITLE
New Help Center

### ISSUE THREAD

**Issue by aaronsamuel137** [phetsims/website#288]
Here's the screenshots for the new help center:

![help-center-with-icons](https://cloud.githubusercontent.com/assets/3193912/10525670/3af05f6e-7343-11e5-8231-018cc1a71def.png)

**Comment by oliver-phet** [comment]
Requirements for admin interface: 
Ability to add/delete a new questions
Ability to edit questions
Ability to write HTML into the interface

**Comment by amanda-phet** [comment]
The screenshots you included above have two that should not be there. Only the first three are accurate- not sure how the last two got included.

**Comment by aaronsamuel137** [comment]
They were all in the same file, but we are aware that it is only the top three that we should go by. Thanks for making sure!

**Comment by oliver-phet** [comment]
Hey @aaronsamuel137 and @mattpen, we will likely be changing the color of the icons I sent you to match an overall website color scheme. Is there a way to program the icons so that the colors could be easily changed on your end?

**Comment by aaronsamuel137** [comment]
>Is there a way to program the icons so that the colors could be easily changed on your end?

There might be a way to do this. I know with font-awesome images you can change the css color and the icon color will change. We'll look into it.

**Comment by oliver-phet** [comment]
> There might be a way to do this. I know with font-awesome images you can change the css color and the icon color will change. We'll look into it.

It is low priority - we can always recreate the .svg images with the new color.

**Comment by mattpen** [comment]
@oliver-phet, if you send the html color code or RGB value we can make this change relatively quickly.

**Comment by oliver-phet** [comment]
Ok, once we have the colors figured out, I’ll let you know.
--
Oliver Nix
PhET Interactive Simulations
University of Colorado
oliver.nix@colorado.edu<mailto:oliver.nix@colorado.edu>

On Oct 22, 2015, at 1:08 PM, Matt Pennington <notifications@github.com<mailto:notifications@github.com>> wrote:


@oliver-phet<https://github.com/oliver-phet>, if you send the html color code or RGB value we can make this change relatively quickly.

**Comment by oliver-phet** [comment]
@mattpen Would it be possible for you to put what you have done on this on phet-dev? That way we can take a look!

**Comment by mattpen** [comment]
@oliver-phet Sure thing, I'll push it to phet-dev later tonight.

**Comment by mattpen** [comment]
@oliver-phet I've deployed what I have done so far. 
This includes:
* The main HelpCenter page.
* Individual HelpCenter pages will show questions/answers.
* In the Administration section, there is now a HelpCenter link.  On this page you can add new questions, edit questions or answers, or change the category of a question.  You may enter html tags here.

Still to do:
* Adding images to the specific topic pages
* Adding the Running_sims_box image
* Separating the Running Sims page by system type.

I purposefully did not add all of the questions from the spreadsheet so that we can add/edit them to test the admin interface.

**Comment by mattpen** [comment]
@oliver-phet Can you send the icons for the different systems that will go on the Running Sims page?  So far I just have the blue box that they will sit in.

**Comment by mattpen** [comment]
I'm taking this down from phet-dev to demonstrate some changes from #298.  I'll republish again on request or when I get the running-sims box working.

**Comment by oliver-phet** [comment]
All of those icons are font awesome. I was under the impression you could just load those in since font awesome is already apart of the website?

**Comment by mattpen** [comment]
OK, thanks @oliver-phet. I'll see if I can find them.

**Comment by aaronsamuel137** [comment]
I can help with this if you can't find them @mattpen.

**Comment by mattpen** [comment]
@oliver-phet or @aaronsamuel137 Do either of you know which icon the chromebook image is using? I can't find this one. Would the plain [chrome icon](http://fortawesome.github.io/Font-Awesome/icon/chrome/) work?

**Comment by oliver-phet** [comment]
Yes, let's use the plain chrome icon for now. Thanks!

**Comment by amanda-phet** [comment]
It is supposed to be the chrome icon, so yes use that.

**Comment by mattpen** [comment]
The help center is mostly finished, with the exception of adding the actual q-and-a's.  This will have to wait until deployment.  @aaronsamuel137 currently has the new front-page pushed to phet-dev.  I'll work with him to get the branches merged.

**Comment by oliver-phet** [comment]
Sounds great. Thanks @mattpen

**Comment by mattpen** [comment]
@oliver-phet I had one question.  In the edit-question function, the current behavior is when the Category or System is changed ( for example if you change an offline question to an account question), all associated translations will be deleted.  Is this acceptable?  If not, is it ok if we just don't allow the category to be changed?  
The reasoning is that we have setup the string keys to contain the category and system, and it is not easy to keep the translations available when the name of the key changes.

**Comment by oliver-phet** [comment]
Just discussed - ok to wipe out translations if things move. We don't anticipate category/system changes too frequently.

**Comment by mattpen** [comment]
@oliver-phet The [HelpCenter](https://phet-dev.colorado.edu/en/help-center) is now up on phet-dev.  Can you take a look and let me know if there are any changes needed?

**Comment by oliver-phet** [comment]
Some things that need tweaked:
* Each question should be initially visible, but only the first line of the answer (followed by "..."). When a user clicks the question or answer, the entire answer expands. 
* The top level page (http://phet-dev.colorado.edu/en/help-center) is showing a "_" in the bottom-right corner of the icon when the category is hovered over
* 'Troubleshooting' and 'FAQ' sections can be destroyed!

**Comment by mattpen** [comment]
@oliver-phet 
A couple of questions. 
1. Do you want to remove Troubleshooting and FAQ in the same deploy to production as when we add the helpcenter?  Currently the helpcenter doesn't have any content so there would be a period of time where the info is not available on the site.  If this is a problem we could either post the helpcenter in one deploy and remove Troubleshooting/FAQ on the next, or find some means of uploading the helpcenter content on the initial deploy (this would take some time to setup).
2. In the running sims page, there is also an underline when you hover over the system icons - would you like to remove or change this behavior?

**Comment by oliver-phet** [comment]
1. Can we add the content on phet-dev then publish all to the live site? If so, we can remove Troubleshooting/FAQ at the same time.
2. The underline on the platforms in the Running Sims page is fine. I think it looks odd on the main categories - unless it is there for accessibility reasons.

**Comment by mattpen** [comment]
There is currently no method in place to transfer database content from phet-dev to the main site. Any content you add on phet-dev will not automatically transfer to the live site.  Its certainly possible to make this happen, but it requires additional development time.

It shouldn't be a problem to remove the underlines. I'm not an expert on accessibility issues, but I don't think it would be a problem.  Maybe @jessegreenberg could comment on that?

**Comment by amanda-phet** [comment]
I see a tiny underline on the main categories to the right of the icon in Chrome but not in Firefox. 

Please keep the underlines on the platforms under Running Sims. They don't need to be under the icons, but definitely under the text (for accessibility).

**Comment by oliver-phet** [comment]
I'm not seeing the main category underline on Firefox, either. Seems to be partially present on Safari, but seems to mostly be a Chrome issue.

Can we keep the underlines as @amanda-phet described above? Under text, but not icons?

Also - we can get the help center populated on the live site quickly. Maybe leave Troubleshooting/FAQ then once we have help center populated, we can take those 2 down.

**Comment by mattpen** [comment]
I've removed the underline from the images on the initial navigation page (its not deployed yet).  Removing them from the font awesome icons will take a little more work but I can get that done today.

Would you like to coordinate a time so we can deploy the helpcenter, upload the content, and remove the troubleshooting/faq in the same night?

**Comment by oliver-phet** [comment]
We will likely want to have an undergrad do all of the new content. I can ask around to see if any of them are available in the evening to do this. Would it be tonight?
--
Oliver Nix
PhET Interactive Simulations
University of Colorado
oliver.nix@colorado.edu<mailto:oliver.nix@colorado.edu>

On Nov 11, 2015, at 11:49 AM, Matt Pennington <notifications@github.com<mailto:notifications@github.com>> wrote:


I've removed the underline from the images on the initial navigation page (its not deployed yet). Removing them from the font awesome icons will take a little more work but I can get that done today.

Would you like to coordinate a time so we can deploy the helpcenter, upload the content, and remove the troubleshooting/faq in the same night?

**Comment by mattpen** [comment]
@oliver-phet I'm not available tonight, I could do it tomorrow (Thurs) night or Sunday night.  I'd also like to coordinate with @aaronsamuel137 as this would be my first deploy to the production site.

**Comment by amanda-phet** [comment]
Do we need to do some testing on this before deploying? 

Also, there is still one tweak left that was requested by @oliver-phet : 

>Each question should be initially visible, but only the first line of the answer (followed by "..."). When a user clicks the question or answer, the entire answer expands.

**Comment by jessegreenberg** [comment]
> They don't need to be under the icons, but definitely under the text (for accessibility)

This would have been my suggestion as well.

**Comment by ariel-phet** [comment]
Minor...but I think we want to have the "iO" icon more inline with the "iO" logo

**Comment by amanda-phet** [comment]
@ariel-phet should we ask Dave to redesign that icon? He designed it to fit the look of the other icons but we could make it a "capital" O ourselves if that works for you.

**Comment by oliver-phet** [comment]
I'm going to create a mockup using the iO logo.

**Comment by oliver-phet** [comment]
@mattpen Can you do a little testing of this once you think it is ready? (Enter/edit a new question/answer) On a couple platforms and 3 major browsers (IE, Firefox, Chrome)

**Comment by oliver-phet** [comment]
@mattpen To consolidate here some changes we would like you to make. You can reference the attached screenshot. I will email the new translation icon.
![screen shot 2015-11-11 at 1 37 15 pm](https://cloud.githubusercontent.com/assets/5090275/11102500/29b8a91c-887a-11e5-805e-4f7273c93d01.png)

- [ ] Getting Started category seems to be higher than other categories
- [ ] Include tooltips for category icons
- [ ] Remove IO 
- [ ] Realign 2 bottom icons
- [ ] New translation icon
- [ ] Expandable questions

Plan is to do a round of student testing on phet-dev, then we'll move it to the live site and have a student upload questions.

**Comment by oliver-phet** [comment]
Another issue that needs fixed: when a English help center string is edited, the associated website string should be changed to 'out-of-date' rather than 'up to date' or 'not-translated'.

**Comment by mattpen** [comment]
@oliver-phet I have these items done and I will push to phet-dev tonight:

 Getting Started category seems to be higher than other categories
 Include tooltips for category icons
 Remove IO
 Realign 2 bottom icons
 New translation icon
 Expandable questions

**Comment by mattpen** [comment]
@oliver-phet The updates are up on phet-dev now.  I'm still working on the issue of having to double-click to get the question to expand.

**Comment by mattpen** [comment]
FYI - If you're wondering where the questions went, there was a database sync with the production site over the weekend and the questions were removed.  This is not an issue that will happen once it goes live.

**Comment by mattpen** [comment]
@oliver-phet I fixed the problem with expandable questions this morning and pushed it to phet-dev.  I think the page is ready for testing.

**Comment by oliver-phet** [comment]
I just tested: http://phet-dev.colorado.edu/en/help-center/getting-started

It took me 2 clicks to expand a question. We also will need to cursor to change to a "hand" when hovering over an expandable question. 

Also, the alignment of the icons once you are into a category page, seems off. See the original .png for alignment. They need to be shifted to the right so that, for instance, 'Getting Started' is left-aligned with 'Help Center' and the questions below. The getting started icon will not be left-aligned.

**Comment by phet-steele** [comment]
Side note: took me 2 clicks to expand a question after each page refresh, but only one click to open and close after the original two.

**Comment by mattpen** [comment]
* Which browsers are you using @oliver-phet and @phet-steele?  I'm using Chrome and it only takes one click.
* I've updated the hover cursor to a pointer.
* Oliver, I'm not exactly sure whats out of alignment.  Can you take a screenshot and highlight it?

**Comment by phet-steele** [comment]
Windows 8.1 on both FF and Chrome. Again, it only takes one click after the initial two.

**Comment by oliver-phet** [comment]
Just did a hard refresh and behavior is fixed.
--
Oliver Nix
PhET Interactive Simulations
University of Colorado
oliver.nix@colorado.edu<mailto:oliver.nix@colorado.edu>

On Nov 16, 2015, at 3:09 PM, phet-steele <notifications@github.com<mailto:notifications@github.com>> wrote:


Windows 8.1 on both FF and Chrome. Again, it only takes one click after the initial two.

**Comment by oliver-phet** [comment]
@mattpen Regarding alignment, notice the alignment of the blue 'Getting Started' text:
Currently on phet-dev: 
<img width="1280" alt="screen shot 2015-11-17 at 11 43 23 am" src="https://cloud.githubusercontent.com/assets/5090275/11221111/98f40754-8d20-11e5-9da3-d739fad41a01.png">

Design/requested alignment:
<img width="1280" alt="screen shot 2015-11-16 at 12 14 53 pm" src="https://cloud.githubusercontent.com/assets/5090275/11221116/a111d07e-8d20-11e5-87af-e8132d3b5f43.png">

**Comment by mattpen** [comment]
OK, I see what you mean.  I caught this and corrected it in the code yesterday.  It will show up on phet-dev on the next push.

**Comment by mattpen** [comment]
@oliver-phet - I fixed the alignment of the category icons and published it on phet-dev.

**Comment by oliver-phet** [comment]
Looks good, Matt!

**Comment by mattpen** [comment]
@oliver-phet - I think I've addressed all of the concerns for the help-center, with the exception of the ability to reorder questions.  This will take a little thought to automate, we can do it manually in the meantime.  

Is there anything I've missed? Anything else we need to test before moving it to the main site?

**Comment by oliver-phet** [comment]
Thanks for the update. We are going to play around with the icon spacing and text a bit before we move to the main site. I'd also like the student testers to play around with adding more questions before deploy.

One other question for you - should we plan on adding questions once the help center is on the main site? Or can we move the question content over from dev to main?

**Comment by mattpen** [comment]
@oliver-phet Moving the questions from the dev site to production doesn't have a straight forward solution.  It's probably easier to just copy/paste everything manually than write a script for this.  Here's my proposal for getting the questions up.
1. Once testing is complete we do the initial deploy where only the Admin utility goes live and we keep the links to the HelpCenter pages hidden.  We can take as much time as we need to enter questions and make sure they are setup correctly.
2. After we're confident everything looks good, we do a 2nd deploy where we publish the helpcenter links on the home page and nav menu and simultaneously remove the Troubleshooting/FAQ sections.
How does that sound?

**Comment by oliver-phet** [comment]
That sounds perfect. I hadn't thought of keeping the pages hidden.

**Comment by oliver-phet** [comment]
@mattpen Here are a few tweaks to the look of the help center:

- Realign the icons so that they are right-aligned with CU logo
- Add text below icons
- Add hover-over color for icons (same blue)
- Remove orange 'Help Center' header 

New mockup:
![new-help-center-with-icons](https://cloud.githubusercontent.com/assets/5090275/11672183/852677dc-9dcb-11e5-9236-f459e1a1f9d1.png)

**Comment by mattpen** [comment]
> * Remove orange 'Help Center' header

Is this requirement critical?  It would require making a major change to the website as its currently created by the main Wicket application, not the HelpCenter pages.  I noticed in the #301 New Sims mockup its missing as well.  Is this a change we are going to make to the entire site?  It would be easier to take it down from the entire site than just one page.  

> * Realign the icons so that they are right-aligned with CU logo

The CU Logo moves as the page is resized, but generally the main content does not.  Is it sufficient to align the icons with the logo in maximized position, or should they resize with the window?

The other two items should be pretty simple to implement.

**Comment by mattpen** [comment]
@oliver-phet I found an easy way to remove the orange title.  This change, along with the others, are now up on phet-dev.

**Comment by mattpen** [comment]
@oliver-phet A few questions.
* Do you still want the orange title on the main page (the one with just the five icons)?
* Should the text of the category names on the main page change to cyan (or whatever you call that blue)?
* Do we need hover highlighting on the system icons?

**Comment by oliver-phet** [comment]
We can keep the title on the main page (in orange, like the rest of the site). Can you make black hover highlighting for the system icons?

**Comment by oliver-phet** [comment]
Also - I think the fix to remove the orange title won't work - definitely noticeable that it is there for a second.

**Comment by mattpen** [comment]
I found a different fix for the orange title, and the system icons now have hover highlighting.  This has been pushed to phet-dev.

**Comment by mattpen** [comment]
As there was some confusion, I'm clarifying which of the in-progress website changes are dependent on each other.  
This change is dependent on the sponsor-page.  It will be deployed at the same time.

**Comment by mattpen** [comment]
@oliver-phet - I commited the changes for #333 tonight and it was working.  It was a minor change, so I went ahead and pushed the helpcenter to the main site.  It is only accessible for now in the admin page and in the translation utility.  It is safe to go ahead and start adding the Q/A strings.  

As its structured, there is no way to make the actual pages live without publishing links in the navigation menu, so if we need to test the appearance we can do so on phet-dev (I'll have the development version back by tomorrow morning).

**Comment by oliver-phet** [comment]
@mattpen We discussed at the web meeting. Can you design the 'Running Sims' page to have the design below? (We only want to institute this on one page to minimize development time)
This is modeled after the design for the offline access page: https://drive.google.com/open?id=0Bz7jTh2s-5L5cFJRYzQxcWhqNXc

<img width="1144" alt="screen shot 2016-01-20 at 2 21 43 pm" src="https://cloud.githubusercontent.com/assets/5090275/12463524/2ac47622-bf81-11e5-9b4b-63e7f7fa660c.png">

Additionally, the current questions on the Troubleshooting/FAQ pages have individual links (ie: http://phet.colorado.edu/en/faqs#q2-header) Is it possible to add this type of behavior to the new help center. This functionality is very useful when directing users to specific questions/answers.

**Comment by mattpen** [comment]
> We only want to institute this on one page to minimize development time

The development time to institute this on all pages instead of just one is basically negligible.  It's just copy/paste at that point.

>  Is it possible to add this type of behavior to the new help center. 

Yes, it is possible. It should be easy to match the behavior on the FAQ page.

**Comment by mattpen** [comment]
@oliver-phet - both of these requests have been posted to phet-dev.  See https://phet-dev.colorado.edu/en/help-center/running-sims/general#q30-header

**Comment by mattpen** [comment]
@oliver-phet - The font/colors of the questions on the running sims page have the same style as the FAQ page.  Should the triangles be blue instead? Should the right-facing triangles have a different color from the downward-facing triangles?

**Comment by oliver-phet** [comment]
@mattpen Both of the triangles (right and down) should be the same blue as the font.

**Comment by amanda-phet** [comment]
@mattpen it might be helpful to model the design after the Offline Access page, not the FAQ/Troubleshooting pages. We decided to go with the <h2> headings here but with a little more spacing between each question.

**Comment by mattpen** [comment]
@oliver-phet, @amanda-phet - I sent an email requesting new artwork for the blue triangles.

Do you want to try this behavior on the other pages (Getting Started, etc.) as well?

**Comment by oliver-phet** [comment]
@mattpen The blue triangles look great. Let's do that on the other pages as well.

**Comment by mattpen** [comment]
The triangle toggle behavior is now applied to all the pages on phet-dev.

**Comment by phet-steele** [comment]
@mattpen all strings look good on my end. I edited things on phet-dev, so check after midnight to see any changes :smile:.
One small problem though phetsims/website/issues/375.

**Comment by mattpen** [comment]
@oliver-phet @phet-steele - Is the help-center ready to go?  Is there anything else I need to fix before we publish it?

**Comment by oliver-phet** [comment]
Take the red pill, Matt!

**Comment by oliver-phet** [comment]
Which also means make the 'Troubleshooting' and 'FAQs' hidden/removed.

**Comment by mattpen** [comment]
This was deployed to figaro tonight!  Closing this issue, I'll consider any further work to be bug fix or improvements and should be discussed in a new issue.
